src/parameterizations/vertical/MOM_geothermal.F90

```diff
@@ -100,6 +102,11 @@ subroutine geothermal(h, tv, dt, ea, eb, G, GV, CS, halo)
   real :: Irho_cp       ! inverse of heat capacity per unit layer volume
                         ! [degC H m2 J-1 ~> degC m3 J-1 or degC kg J-1]
 
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G)) :: temp_old  ! Temperature of each layer before any heat is added, for diagnostics [degC]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G)) :: h_old     ! Thickness of each layer before any heat is added, for diagnostics [m or kg m-2]
+  real, dimension(SZI_(G),SZJ_(G),SZK_(G)) :: work_3d   ! Scratch variable used to calculate change in heat due to geothermal
```

Comment:
These lines exceed 120 characters, so the comments will need to be split over two lines.
(Ideally lines will not exceed 80, but in this case ~100 should be OK.)

(No `!<` is needed here since it's a local variable)